elytron/ELY-2078-encryption-for-FilesystemSecurityRealm.adoc

```diff
@@ -0,0 +1,222 @@
+= [ELY-2078] Add encryption and integrity support to FilesystemSecurityRealm
+:author:            Ashpan Raskar
+:email:             araskar@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This objective of this proposal is to securely encrypt the identities and credentials in the `FileSystemSecurityRealm` stored on the local filesystem. In the past, the default for storing identities was via property files, but in future versions of WildFly, the plan is to use a FileSystemSecurityRealm as the default out of the box configuration instead. The FileSystemSecurityRealm currently uses Base64 to encode clear passswords, hashed passwords, and attributes, but we want to add the ability to encrypt the credentials and attributes for an identity. In order to do this, we plan to add SecretKey support, which is a symmetric key to encrypt and decrypt data. We wish to also add the ability to verify integrity for the filesystem realm as well, this will be done with the use of a Public and Private Key pair, that will create a signature on every identity at the bottom of the xml file.
```

Comment:
> the plan is to use a FileSystemSecurityRealm as the default out of the box configuration instead

Is this in the scope of this RFE?

Reply:
Can you add this info the Non-Requirements section please? It would be also helpful having the follow-up RFE linked on the Jira issue.

Reply:
Added